soc/arc/snps_qemu/CMakeLists.txt

```diff
@@ -1,5 +1,18 @@
 # SPDX-License-Identifier: Apache-2.0
 
-if(CONFIG_ISA_ARCV2)
-  zephyr_compile_options(-mno-sdata)
+if(COMPILER STREQUAL gcc)
+	zephyr_compile_options(-mcpu=${GCC_M_CPU})
+
+
+	if(CONFIG_ISA_ARCV2)
+	  zephyr_compile_options(-mno-sdata)
+	endif()
+else()
+
+	zephyr_compile_options( -arcv2hs -core2 -Xatomic
```

Comment:
This shoulde be something like `zephyr_compile_options_ifdef(CONFIG_SOC_QEMI_ARC_HS ...` as currently we only add options for one configuration. Same for zephyr_ld_options, you can use https://github.com/zephyrproject-rtos/zephyr/blob/dca33a126d1765cb647c10eabeea36eafcbd0f92/soc/arc/snps_nsim/CMakeLists.txt#L52 as reference.

Reply:
Done